sdk/python/feast/saved_dataset.py

```diff
@@ -203,21 +206,106 @@ def to_arrow(self) -> pyarrow.Table:
 
         return self._retrieval_job.to_arrow()
 
-    def as_reference(self, profiler: "Profiler") -> "ValidationReference":
-        return ValidationReference(profiler=profiler, dataset=self)
+    def as_reference(self, name: str, profiler: "Profiler") -> "ValidationReference":
+        return ValidationReference.from_saved_dataset(
+            name=name, profiler=profiler, dataset=self
+        )
 
     def get_profile(self, profiler: Profiler) -> Profile:
         return profiler.analyze_dataset(self.to_df())
 
 
 class ValidationReference:
-    dataset: SavedDataset
+    name: str
+    dataset_name: str
     profiler: Profiler
 
-    def __init__(self, dataset: SavedDataset, profiler: Profiler):
-        self.dataset = dataset
+    _profile: Optional[Profile] = None
+    _dataset: Optional[SavedDataset] = None
+
+    def __init__(self, name: str, dataset_name: str, profiler: Profiler):
```

Comment:
We should add description and tags here, since this is effectively a new top level object in feast?